### ISSUE TITLE
Infobase

### ISSUE THREAD

**Issue by hoyes** [camdram/camdram#21]
Create a area of the site which contains free-form user-editable content and import data from existing infobase.

**Comment by hoyes** [comment]
I had a thought about this last night, in light of work in #198. I think revision/roll-back stuff was originally the main reason for including the CMF stuff. I wondered whether using the method of revision tracking used in #198 would be significantly simpler, as all it requires is a few annotations to a Doctrine entity, so it would end up very similar to the show/society/venue code.

The only slightly tricky bit would be copying the v1 revision history over to a new table.

We could also potentially separate out the task of moving from custom markdown to WYSIWYG/HTML and tackle that as a separate issue (maybe post dropping support for v1).

**Comment by hoyes** [comment]
In the name of making progress on this, I've made a very basic read-only view of the Infobase above, along with a very basic script to import from the old tables (`camdram:infobase:import`). In order to simplify things a bit, I have:
 * Stuck with Camdram mark-up (HTML/WYSIWYG can happen at some point in the future)
 * Used named "tags" for each as opposed to a hierarchical tree structure. The import script auto-creates one or more tags for each article based on its position in the tree, which works surprisingly well but it'll likely need some

I'm doing this work in the "infobase" branch so that we can do releases for other work before this is finished.

Still to do though:
 - [ ] Edit/add article forms
 - [ ] Make import console command import full revision history correctly
 - [ ] Search integration
 - [ ] The ability to view and roll back to previous revisions
 - [ ] Add missing parts of the "Camdram markup" parser (on importing it looks like there are few things missing, like using "+Header Text+" for headers)
 - [ ] Make legacy URLs redirect correctly
 - [ ] Add something infobase-related to the home page (e.g. a list of recently updated articles?)
 - [ ] Make Infobase homepage and tag listings look nicer

I'm going to start chugging away at the above, but post something here if you want to tackle something - most of the above can be worked on in parallel.

**Comment by hoyes** [comment]
- [ ] Some extra work required to make [CAMDRAM:....] markdown links work (internal links within infobase)

**Comment by hoyes** [comment]
WIP at http://tiberius.camdram.net/infobase

**Comment by ajf58** [comment]
It's looking good, but /please/ comment your source code.
A not insignificant amount of work was done previously to use the PHPCR to implement the Infobase.
```$ git log -v --follow src/Acts/CamdramLegacyBundle/DataFixtures/PHPCR/LoadPageData.php | less``
Does this work use that code at all, or have we ended up with a confusing mess with two different paths to the same data/views?

**Comment by hoyes** [comment]
The commit above adds lots of comments which I hope helps.

As you probably discovered, there are a lot of extra hoops to jump through for PHPCR (an extra document layer, routing engine, several extra PHP dependencies) so I did this using the Doctrine extensions 'Loggable' thing, which as I explained above is I think the quickest path to a functioning infobase (I was able to do all the above in a day). I used the LoadPageData class as a reference but as PHPCR and Loggable work very differently there wasn't sufficient overlap to justify copy+pasting.

I appreciate your efforts with PHPCR - I'd say the learning experience informed the direction I took above, even if it does turn out to be a dead end. I'm not against ditching the above and using PHPCR still at some point in the future though, e.g. if we eventually move to HTML/WYSIWYG and PHPCR makes this easier. We could perhaps just use one or two components (e.g. 'blocks' and the create-js WYSIWYG editor integration) rather than integrate the whole thing.

**Comment by CHTJonas** [comment]
The email conversation on webteam and the minutes of the following CUADC meetings should be taken into account with this issue:
- [05/05/2018](https://docushare.cuadc.org/meeting-minutes/committee_minutes_2018_05_05.pdf)
- [20/05/2018](https://docushare.cuadc.org/meeting-minutes/committee_minutes_2018_05_20.pdf)
- [13/06/2018](https://docushare.cuadc.org/meeting-minutes/committee_minutes_2018_06_13.pdf)
- [20/06/2018](https://docushare.cuadc.org/meeting-minutes/committee_minutes_2018_06_20.pdf)

The CUADC wiki WIP is online at https://wiki.cuadc.org however it hasn't been officially launched and there are some licensing concerns regarding the v1 info base content that appears in there. As they talk about approaching Camdram for hosting (currently they're on the SRCF) I am going to send an informal email to their webmaster to discuss. There is also a few incorrect points in their minutes I'd like to clarify briefly just so everyone's on that same page.

In the last few months, the links to the Camdram infobase were removed (and v1 decommission in its entirety). Is consensus that the Camdram infobase should be depreciated in favour of a public wiki instance?

**Comment by hoyes** [comment]
I agree a dedicated Wiki is the right approach now.

I do think there is a strong positive case to be made for suggesting bringing the hosting/maintenance side of such a Wiki under the Camdram umbrella. Mainly:
 * Camdram has it's own recruitment issues, but if a server issue were to happen today there are probably at least twice the number of people willing and able to respond on the Camdram team than on CUADC committee.
 * It operates from a "society-neutral" position.
 * Using the Camdram login system would mean the Wiki can reach people that Raven cannot (townies, ARU).
 * A general attitude towards long-term stability that would protect it from the potential whims of future committees.
 * The genesis of this is in some ways related to the v1 Infobase.
 * The Brimir thing for support has proved quite a nice model for taking advantage of third party solutions, but integrated with Camdram using OAuth.

But if the wind is sailing the other way then that's also fine by me!

There seems to be an emphasis in the minutes about assigning CUADC committee members to curate content. That's all a good thing (regardless what happens with hosting), and some sort of "critical mass" of content is probably needed. But if it were me, I wouldn't be so cautious about launching it prior to that - the whole point of a wiki is to democraticise the content creation process.

We should perhaps manage expectations a little about hosting image files. I'm not sure what sort of controls MediaWiki offers but our offering would probable either involve media uploads being disabled or limited to a certain resolution. It looks like Mediawiki has an S3 plugin so there might be altnerative options there...

**Comment by hoyes** [comment]
Trying to avoid getting carried away while discussions are still happening, but MediaWiki has an API (https://www.mediawiki.org/wiki/API:Main_page) that would maybe allow some of the former ideas to still be implemented. E.g. cross-linking a Camdram venue page with the relevant Wiki page, maybe with a content preview on Camdram or something.

**Comment by CHTJonas** [comment]
> It looks like Mediawiki has an S3 plugin so there might be altnerative options there...

We could combine a hybrid approach where images/files are hosted on a [Minio](http://minio.io) server running on the SRCF whilst the actual wiki is served from Camdram? Or just go the whole way and host media out of an AWS bucket with organisation(s) X (and Y) fronting the costs...

**Comment by CHTJonas** [comment]
Incidentally, Tiberius appears to have a 30G unused second hard drive:
```
sudo lsblk
NAME    MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
xvda    202:0    0  20G  0 disk
├─xvda1 202:1    0  19G  0 part /
└─xvda2 202:2    0   1G  0 part [SWAP]
xvdb    202:16   0  30G  0 disk
```

**Comment by philosophicles** [comment]
> Is consensus that the Camdram infobase should be depreciated in favour of a public wiki instance?

To me that's a slightly wrong question. Without wanting to sound too cliche: the Infobase is a concept not a specific implementation. 

Even the name "Infobase" is just a label, not a definition; when Camdram and the Infobase were originally created, Wikipedia was barely off the ground, the concept of wikis wasn't familiar to most people, and thus the Infobase needed a descriptive name of its own. It was _always_ a public wiki instance, albeit one custom-coded from scratch rather than using third-party platforms/frameworks. If Camdram had been born just a few years later, the Infobase would probably have been called the Camdram Wiki from day 1. 

The question of whether we are happy to stop "competing" with another society for provision of an Infobase concept is a valid one - whether to let someone else take the reins since they've done most of the legwork recently. Personally, like Peter, I'd like to see this remain under Camdram-neutral control (and fully Open content) but that's just my opinion.

The decision to go with a mature third-party wiki platform rather than a new implementation from scratch seems like a no-brainer though :)

**Comment by CHTJonas** [comment]
I sent an email to CUADC a few days ago and am waiting to hear back 📧👂

Will update with progress on here or on webteam as appropriate

**Comment by CHTJonas** [comment]
Have had a reply on behalf of CUADC committee and whilst they acknowledge the benefits you list Peter, basically they'd like to keep things the way they are (ie. themselves hosting) at least for the moment.

Their main concern/justification is that they will be adding lots of material to the wiki over summer and that it's an active project of theirs. They're concerned about the image that will be portrayed if they are too active/dominant on a society-neutral website. They also think that to get more people to start using and contributing to it, it's better there's a society/committee publicizing and pushing it.

They do say however that they might ask to move the wiki over to Camdram when it settles down some time in the future because they feel like long term that's the right thing to do.

**Comment by philosophicles** [comment]
One question I have - is this CUADC wiki covering all the same broad areas that the Infobase covered, or has its scope been changed/moved/reduced in any way? (I cannot log in to the work-in-progress due to lack of Raven.)

I'm wondering if they're still including topics like "about each venue" - the section that had pretty detailed technical data about what each venue in Cambridge provided, dimensions, lighting/sound equipment, etc. 

I could imagine that this new CUADC effort might focus more on the sections that detailed the different creative/technical roles on a typical show, how to get involved, etc. 

If any major sections have been dropped as "out of scope" (rather than deprecated because they just aren't relevant any more) then it might be good to know about that in case we can help fill the gaps. The venue technical data part specifically would relate to #97 if it's not in the new Wiki.

**Comment by hoyes** [comment]
I agree that it's fantastic that CUADC are being proactive in resurrecting this after some years of neglect by Camdram. I wonder whether we should maybe lean on CUADC a little on a couple of points, both of which were principles of the former Infobase:
 1. Open access - readable by anyone without a login, and editable by anyone regardless of possession of a Raven account (relevant for former students and non-Cambridge uni students).
 2. Content licensing - content derived from the former Infobase quoted https://creativecommons.org/licenses/by-nc-sa/2.5/ (https://github.com/camdram/camdram-v1/blob/8179c3d7d6f05497bbc24ba351b0d44938e2fe27/index.php#L170), and I think applying a similar licence to content on the CUADC wiki would manage expectations about how content people submit to the CUADC wiki today might be reused/transferred in the future. This should really be fait accompli for any content derived from the former Infobase.

Would they consider maybe adopting https://www.mediawiki.org/wiki/Extension:OAuth2_Client to Camdram as a bit of a compromise, and quoting the same Creative Commons licence (or maybe a more recent version) in the footer? This may come across as slightly pedantic, but I think both these steps would greatly simplify transferring anything to Camdram at a later date, if the situation aroused. Plus using Camdram for authentication might, symbolically, emphasize to people at large that Camdram endorses their efforts.

**Comment by CHTJonas** [comment]
Yeah I agree with everything that's been said above. @zzaanna I was thinking of emailing you regarding this but maybe it's just easier to mention you here instead. What are your/committee's opinion on the two above points? Both should be super easy as MediaWiki plugins are a dream to install & setup.

I think that since the wiki does contain some material that was mirrored from the info base before it was shutdown, legally it needs to have the correct CCL license on it. That being said, it's quite pedantic and beyond sending a message, nobody is about to get sued for copyright infringement! 🤣

**Comment by hoyes** [comment]
My potted summary of CCL:
 * Copyright remains collectively with the original contributor(s) to the article
 * Contributors are required to grant an irrevocable licence to others to reuse and/or adapt the content in any medium, so long as the CCL licence is retained (and therefore the derivative work has the same conditions).

So, while a little pedantic today I think this is quite important if you take a medium/long term view. E.g.
 * If the CUADC wiki were to fall into disrepair and off the Internet in a couple of years' time, anyone else would be free to take the content and publish it elsewhere without complications under the CCL [heavy wink, ironic nod...]
 * Ditto, attempts by a future nefarious committee to restrict access to payed-up CUADC members, or heavily moderate content to enforce a certain point of view or something would be ineffective.
 * Means a contributor today can't come back to bite a committee in the future by changing the rights granted on a past contribution or something. For people spending their time writing new content today, it's probably worth spending a half a second thinking about the copyright/licensing thing.

More on the fence about OAuth. We're probably not quite there yet, but the idea is that Camdram is trying to solve the single identity/SSO (single sign-on) problem for Cambridge theatre peeps in one place so others don't have to.

**Comment by hoyes** [comment]
I should probably echo @CHTJonas too that this isn't passing any judgment on anyone involved in this discussion. And chances are that future committees will do the right thing too. I'm just trying to convince people that it's a small but helpful thing to put in place today to mitigate against the 2% probability of complications in the future ;)

**Comment by zzaanna** [comment]
Oh yes happy to implement both and I've okayed it with the committee.

For 1, yes it's definitely something we plan to do. Actually a huge life saver so I don't have to worry about logging in through ARU/cantab or adding anything new. Will talk to Charlie about details.

For 2, I'll quickly add a footnote:
> This work is licensed under a [Creative Commons Attribution-NonCommercial-ShareAlike 4.0 Generic License](https://creativecommons.org/licenses/by-nc-sa/4.0/). By using this site, you acknowledge that you have read and agree to the Privacy Policy and Disclaimer. 

Thank you very much for all your help! I've never seen how old infobase works and know zilch about Licensing and stuff so all the input are much appreciated!

**Comment by hoyes** [comment]
That all sounds great. I'm going to close this ticket as it looks like Camdram's work is done here. There are a few idiosyncrasies with setting up OAuth at the moment which we hope to fix but @CHTJonas  should be all too familiar with in the mean time...

If yourself or anyone on CUADC need to get in touch about this (or indeed anything else Camdram-related in the future) then https://gitter.im/camdram/development is probably a good place to find one of us.

Look forward to seeing the Wiki go live.
